collector/cpu_freebsd.go

```diff
@@ -115,11 +122,18 @@ func (c *statCollector) Update(ch chan<- prometheus.Metric) (err error) {
 		return err
 	}
 	for cpu, t := range cpuTimes {
-		ch <- c.cpu.mustNewConstMetric(float64(t.user), strconv.Itoa(cpu), "user")
-		ch <- c.cpu.mustNewConstMetric(float64(t.nice), strconv.Itoa(cpu), "nice")
-		ch <- c.cpu.mustNewConstMetric(float64(t.sys), strconv.Itoa(cpu), "system")
-		ch <- c.cpu.mustNewConstMetric(float64(t.intr), strconv.Itoa(cpu), "interrupt")
-		ch <- c.cpu.mustNewConstMetric(float64(t.idle), strconv.Itoa(cpu), "idle")
+		lcpu := strconv.Itoa(cpu)
+		ch <- c.cpu.mustNewConstMetric(float64(t.user), lcpu, "user")
+		ch <- c.cpu.mustNewConstMetric(float64(t.nice), lcpu, "nice")
+		ch <- c.cpu.mustNewConstMetric(float64(t.sys), lcpu, "system")
+		ch <- c.cpu.mustNewConstMetric(float64(t.intr), lcpu, "interrupt")
+		ch <- c.cpu.mustNewConstMetric(float64(t.idle), lcpu, "idle")
+
+		temp, err := unix.SysctlUint32(fmt.Sprintf("dev.cpu.%d.temperature", cpu))
+		if err == nil {
```

Comment:
Personally I'd expect such information as debug output to understand why metrics are missing.

Ideally we could distinguish temporary errors from unsupported errors. Exporting NaN would make the former case a lot clearer, but not setting the metric at all makes probably more sense in the second case.